### PR TITLE
Upgrade wUSDM to restrict USA transfers

### PR DESCRIPTION
**Pull request by fedealconada**
# Description

Upgrades wUSDM to allow transfers everywhere except for Kinto Wallets with USA trait (`840`) 

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Deployment
- [ ] Forge Script
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: src/tokens/bridged/BridgedWUSDM.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+pragma solidity ^0.8.18;
+
+import {BridgedToken} from "./BridgedToken.sol";
+import {IKintoID} from "@kinto-core/interfaces/IKintoID.sol";
+import {IKintoWallet} from "@kinto-core/interfaces/IKintoWallet.sol";
+import {IKintoWalletFactory} from "@kinto-core/interfaces/IKintoWalletFactory.sol";
+
+/**
+ * @title BridgedToken
```
Comment (ylv-io) [inline src/tokens/bridged/BridgedWUSDM.sol:11]:
```suggestion
 * @title BridgedWusdm
```

File: src/tokens/bridged/BridgedWUSDM.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * @notice Implements an ERC20 token with bridging capabilities for cross-chain asset transfers.
+ * Extends OpenZeppelin's ERC20, ERC20Permit, and AccessControl.
+ * @dev Introduces `mint` and `burn` functions secured with the `MINTER_ROLE` for bridging processes.
+ * Inherits ERC20 functionality, permit mechanism for gasless transactions, and role-based access control.
+ */
```
Comment (ylv-io) [inline src/tokens/bridged/BridgedWUSDM.sol:16]: Let's do @notice specific to BridgedWusdm, not just BridgedToken.

File: src/tokens/bridged/BridgedWUSDM.sol [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    constructor(uint8 decimals_) BridgedToken(decimals_) {}
+
+    error TransferNotAllowed(address from, address to, uint256 countryId);
+
+    /**
+     * @dev Transfers a `value` amount of tokens from `from` to `to`, or alternatively mints (or burns) if `from`
+     * (or `to`) is the zero address. All customizations to transfers, mints, and burns should be done by overriding
+     * this function.
+     *
+     * If destination is a KintoWallet, only allow the transfer if
```
Comment (ylv-io) [inline src/tokens/bridged/BridgedWUSDM.sol:32]: `transfer if...`?

**Review by ylv-io — APPROVED** [review]
Great work!

File: src/tokens/bridged/BridgedWUSDM.sol (hunk 3 above)
Comment (fedealconada) [reply src/tokens/bridged/BridgedWUSDM.sol:32]: lol

**Pull request merged**